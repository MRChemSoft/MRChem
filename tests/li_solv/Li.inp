# vim:syntax=sh:
world_prec = 1.0e-3               # Overall relative precision
world_size = 5                    # Size of simulation box 2^n

Printer {
  print_level = 0                       # Level of detail in the printed output
}

MPI {
  numerically_exact = true        # Guarantee identical results in MPI
}

Molecule {
  charge = 1
$coords
Li   0.0     0.0    0.0
$end
}

WaveFunction {
  method = pbe0                     # Wave function method (HF or DFT)
  restricted = true
}
Environment {
  run_SCRF = true
  run_variational = false
  kain_history = 5
  max_iter = 100
  run_hybrid = true
  extrapolate_Vr = true

  Cavity {
    cavity_width = 0.5
    $coords
    0.0     0.0    0.0  4.0
    $end
  }

  Permittivity {
  epsilon_in = 1.0
  epsilon_out = 2.0
  linear = false
  }
}

SCF {
  run = true
  kain = 5                        # Length of KAIN iterative history
  orbital_thrs = 1.0e-3           # Convergence threshold in orbital residual
  guess_type = mw                 # Type of initial guess: none, mw, gto
}
