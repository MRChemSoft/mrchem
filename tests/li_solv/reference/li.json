{
  "input": {
    "molecule": {
      "cavity_coords": [
        {
          "center": [
            0.0,
            0.0,
            0.0
          ],
          "radius": 4.0
        }
      ],
      "cavity_width": 0.5,
      "charge": 1,
      "coords": [
        {
          "atom": "li",
          "xyz": [
            0.0,
            0.0,
            0.0
          ]
        }
      ],
      "multiplicity": 1
    },
    "mpi": {
      "bank_size": -1,
      "numerically_exact": true,
      "shared_memory_size": 10000
    },
    "mra": {
      "basis_order": 5,
      "basis_type": "interpolating",
      "boxes": [
        2,
        2,
        2
      ],
      "corner": [
        -1,
        -1,
        -1
      ],
      "max_scale": 20,
      "min_scale": -4
    },
    "printer": {
      "file_name": "li.inp",
      "print_level": 0,
      "print_mpi": false,
      "print_prec": 6,
      "print_width": 75
    },
    "rsp_calculations": {},
    "scf_calculation": {
      "fock_operator": {
        "coulomb_operator": {
          "poisson_prec": 0.001,
          "shared_memory": false
        },
        "exchange_operator": {
          "exchange_prec": -1.0,
          "poisson_prec": 0.001
        },
        "kinetic_operator": {
          "derivative": "abgv_55"
        },
        "nuclear_operator": {
          "proj_prec": 0.001,
          "shared_memory": false,
          "smooth_prec": 0.001
        },
        "reaction_operator": {
          "accelerate_Vr": true,
          "algorithm": "scrf",
          "convergence_criterion": "dynamic",
          "density_type": "total",
          "epsilon_in": 1.0,
          "epsilon_out": 2.0,
          "formulation": "exponential",
          "kain": 5,
          "max_iter": 100,
          "poisson_prec": 0.001
        },
        "xc_operator": {
          "shared_memory": false,
          "xc_functional": {
            "cutoff": 0.0,
            "functionals": [
              {
                "coef": 1.0,
                "name": "pbe0"
              }
            ],
            "spin": false
          }
        }
      },
      "initial_guess": {
        "file_CUBE_a": "cube_vectors/CUBE_a_vector.json",
        "file_CUBE_b": "cube_vectors/CUBE_b_vector.json",
        "file_CUBE_p": "cube_vectors/CUBE_p_vector.json",
        "file_basis": "initial_guess/mrchem.bas",
        "file_chk": "checkpoint/phi_scf",
        "file_gto_a": "initial_guess/mrchem.moa",
        "file_gto_b": "initial_guess/mrchem.mob",
        "file_gto_p": "initial_guess/mrchem.mop",
        "file_phi_a": "initial_guess/phi_a_scf",
        "file_phi_b": "initial_guess/phi_b_scf",
        "file_phi_p": "initial_guess/phi_p_scf",
        "localize": false,
        "method": "DFT (PBE0)",
        "prec": 0.001,
        "relativity": "None",
        "restricted": true,
        "screen": 12.0,
        "type": "sad",
        "zeta": 2
      },
      "properties": {
        "dipole_moment": {
          "dip-1": {
            "operator": "h_e_dip",
            "precision": 0.001,
            "r_O": [
              0.0,
              0.0,
              0.0
            ]
          }
        }
      },
      "scf_solver": {
        "checkpoint": false,
        "derivative": "abgv_55",
        "energy_thrs": -1.0,
        "file_chk": "checkpoint/phi_scf",
        "final_prec": 0.001,
        "helmholtz_prec": -1.0,
        "kain": 5,
        "light_speed": -1.0,
        "localize": false,
        "max_iter": 100,
        "method": "DFT (PBE0)",
        "orbital_thrs": 0.1,
        "proj_prec": 0.001,
        "relativity": "None",
        "rotation": 0,
        "shared_memory": false,
        "smooth_prec": 0.001,
        "start_prec": 0.001
      }
    },
    "schema_name": "mrchem_input",
    "schema_version": 1
  },
  "output": {
    "properties": {
      "center_of_mass": [
        0.0,
        0.0,
        0.0
      ],
      "charge": 1,
      "dipole_moment": {
        "dip-1": {
          "magnitude": 2.5407415880983347e-13,
          "r_O": [
            0.0,
            0.0,
            0.0
          ],
          "vector": [
            0.0,
            0.0,
            0.0
          ],
          "vector_el": [
            0.0,
            0.0,
            0.0
          ],
          "vector_nuc": [
            0.0,
            0.0,
            0.0
          ]
        }
      },
      "geometry": [
        {
          "symbol": "Li",
          "xyz": [
            0.0,
            0.0,
            0.0
          ]
        }
      ],
      "multiplicity": 1,
      "orbital_energies": {
        "energy": [
          -2.210659505993009
        ],
        "occupation": [
          2.0
        ],
        "spin": [
          "p"
        ],
        "sum_occupied": -4.421319011986018
      },
      "scf_energy": {
        "E_ee": 3.3576788742677035,
        "E_eext": 0.0,
        "E_el": -7.127946930407876,
        "E_en": -16.4914760516857,
        "E_kin": 7.584212118791978,
        "E_next": 0.0,
        "E_nn": 0.0,
        "E_nuc": -0.19212873389394816,
        "E_tot": -7.320075664301824,
        "E_x": -0.4195941455371656,
        "E_xc": -1.2868490616228483,
        "Er_el": 0.12808133537815772,
        "Er_nuc": -0.19212873389394816,
        "Er_tot": -0.06404739851579039
      }
    },
    "provenance": {
      "creator": "MRChem",
      "mpi_processes": 1,
      "nthreads": 1,
      "routine": "mrchem.x",
      "total_cores": 1,
      "version": "1.1.0-alpha"
    },
    "rsp_calculations": null,
    "scf_calculation": {
      "initial_energy": {
        "E_ee": 3.3327993838943355,
        "E_eext": 0.0,
        "E_el": -7.222649094979323,
        "E_en": -16.530755287695392,
        "E_kin": 7.675564301900276,
        "E_next": 0.0,
        "E_nn": 0.0,
        "E_nuc": 0.0,
        "E_tot": -7.222649094979323,
        "E_x": -0.4164844953176007,
        "E_xc": -1.2837729977609427,
        "Er_el": 0.0,
        "Er_nuc": 0.0,
        "Er_tot": 0.0
      },
      "scf_solver": {
        "converged": true,
        "cycles": [
          {
            "energy_terms": {
              "E_ee": 3.3576788742677035,
              "E_eext": 0.0,
              "E_el": -7.127946930407876,
              "E_en": -16.4914760516857,
              "E_kin": 7.584212118791978,
              "E_next": 0.0,
              "E_nn": 0.0,
              "E_nuc": -0.19212873389394816,
              "E_tot": -7.320075664301824,
              "E_x": -0.4195941455371656,
              "E_xc": -1.2868490616228483,
              "Er_el": 0.12808133537815772,
              "Er_nuc": -0.19212873389394816,
              "Er_tot": -0.06404739851579039
            },
            "energy_total": -7.320075664301824,
            "energy_update": 0.0974265693225016,
            "mo_residual": 0.06037084934246604,
            "wall_time": 11.150312992
          }
        ],
        "wall_time": 11.155897119
      },
      "success": true
    },
    "schema_name": "mrchem_output",
    "schema_version": 1,
    "success": true
  }
}
