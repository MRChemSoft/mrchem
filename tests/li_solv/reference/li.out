                                                                           
                                                                           
***************************************************************************
***                                                                     ***
***                                                                     ***
***          __  __ ____   ____ _                                       ***
***         |  \/  |  _ \ / ___| |__   ___ _ __ ___                     ***
***         | |\/| | |_) | |   | '_ \ / _ \ '_ ` _ \                    ***
***         | |  | |  _ <| |___| | | |  __/ | | | | |                   ***
***         |_|  |_|_| \_\\____|_| |_|\___|_| |_| |_|                   ***
***                                                                     ***
***         VERSION            1.1.0-alpha                              ***
***                                                                     ***
***         Git branch         molecule-class-cleanup-input-parser          ***
***         Git commit hash    d073311b4510a57f986e                     ***
***         Git commit author  Anders Brakestad                         ***
***         Git commit date    Mon Mar 28 13:46:33 2022 +0200           ***
***                                                                     ***
***         Contact: luca.frediani@uit.no                               ***
***                                                                     ***
***         Radovan Bast            Magnar Bjorgve                      ***
***         Roberto Di Remigio      Antoine Durdek                      ***
***         Luca Frediani           Gabriel Gerez                       ***
***         Stig Rune Jensen        Jonas Juselius                      ***
***         Rune Monstad            Peter Wind                          ***
***                                                                     ***
***************************************************************************

---------------------------------------------------------------------------

 MPI processes           :       (no bank)                               1
 OpenMP threads          :                                               1
 Total cores             :                                               1
                                                                           
---------------------------------------------------------------------------

XCFun DFT library Copyright 2009-2020 Ulf Ekstrom and contributors.
See http://dftlibs.org/xcfun/ for more information.

This is free software; see the source code for copying conditions.
There is ABSOLUTELY NO WARRANTY; not even for MERCHANTABILITY or
FITNESS FOR A PARTICULAR PURPOSE. For details see the documentation.
Scientific users of this library should cite
U. Ekstrom, L. Visscher, R. Bast, A. J. Thorvaldsen and K. Ruud;
J.Chem.Theor.Comp. 2010, DOI: 10.1021/ct100117s

---------------------------------------------------------------------------

 MRCPP version         : 1.4.0-alpha
 Git branch            : HEAD
 Git commit hash       : 2797a7a18efb69942a6e
 Git commit author     : Stig Rune Jensen
 Git commit date       : Fri Mar 12 12:48:07 2021 +0100

 Linear algebra        : EIGEN v3.3.7
 Parallelization       : NONE

---------------------------------------------------------------------------


                                                                           
===========================================================================
                         MultiResolution Analysis
---------------------------------------------------------------------------
 polynomial order      : 5
 polynomial type       : Interpolating
---------------------------------------------------------------------------
 total boxes           : 8
 boxes                 : [          2           2           2 ]
 unit lengths          : [   16.00000    16.00000    16.00000 ]
 scaling factor        : [    1.00000     1.00000     1.00000 ]
 lower bounds          : [  -16.00000   -16.00000   -16.00000 ]
 upper bounds          : [   16.00000    16.00000    16.00000 ]
 total length          : [   32.00000    32.00000    32.00000 ]
===========================================================================


                                                                           
***************************************************************************
***                                                                     ***
***                        Initializing Molecule                        ***
***                                                                     ***
***************************************************************************
                                                                           
                                                                           
===========================================================================
                                 Molecule
---------------------------------------------------------------------------
 Charge                  :                                               1
 Multiplicity            :                                               1
---------------------------------------------------------------------------
    N    Atom            :               x               y               z
---------------------------------------------------------------------------
    0      Li            :        0.000000        0.000000        0.000000
---------------------------------------------------------------------------
 Center of mass          :        0.000000        0.000000        0.000000
===========================================================================


                                                                           
***************************************************************************
***                                                                     ***
***                 Computing Ground State Wavefunction                 ***
***                                                                     ***
***************************************************************************
                                                                           
                                                                           
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 Calculation             : Compute initial orbitals
 Method                  : Diagonalize SAD Hamiltonian
 Precision               : 1.00000e-03
 Screening               : 1.20000e+01 StdDev
 Restricted              : True
 Functional              : LDA (SVWN5)
 AO basis                : Hydrogenic orbitals
 Zeta quality            : 2
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


===========================================================================
                            Molecular Orbitals
---------------------------------------------------------------------------
 Alpha electrons         :                                               1
 Beta electrons          :                                               1
 Total electrons         :                                               2
---------------------------------------------------------------------------
    n  Occ Spin          :                                            Norm
---------------------------------------------------------------------------
    0    2    p          :                              1.000000000000e+00
===========================================================================


~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 Calculation             : Compute initial energy
 Method                  : DFT (PBE0)
 Relativity              : None
 Precision               : 1.00000e-03
 Localization            : Off
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


===========================================================================
                        Molecular Energy (initial)
---------------------------------------------------------------------------
 Kinetic energy          :            (au)                  7.675564301900
 E-N energy              :            (au)                -16.530755287695
 Coulomb energy          :            (au)                  3.332799383894
 Exchange energy         :            (au)                 -0.416484495318
 X-C energy              :            (au)                 -1.283772997761
 Ext. field (el)         :            (au)                  0.000000000000
---------------------------------------------------------------------------
 N-N energy              :            (au)                  0.000000000000
 Ext. field (nuc)        :            (au)                  0.000000000000
 Tot. Reac. Energy       :            (au)                  0.000000000000
 El. Reac. Energy        :            (au)                  0.000000000000
 Nuc. Reac. Energy       :            (au)                  0.000000000000
---------------------------------------------------------------------------
 Electronic energy       :            (au)                 -7.222649094979
 Nuclear energy          :            (au)                  0.000000000000
---------------------------------------------------------------------------
 Total energy            :            (au)             -7.222649094979e+00
                         :      (kcal/mol)             -4.532280698364e+03
                         :        (kJ/mol)             -1.896306245426e+04
                         :            (eV)             -1.965382856045e+02
===========================================================================


===========================================================================
                        Orbital Energies (initial)
---------------------------------------------------------------------------
    n  Occ Spin          :                                         Epsilon
---------------------------------------------------------------------------
    0    2    p          :            (au)                 -2.332093141176
---------------------------------------------------------------------------
 Sum occupied            :            (au)                 -4.664186282352
===========================================================================


~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 Calculation             : Optimize ground state orbitals
 Method                  : DFT (PBE0)
 Relativity              : None
 Checkpointing           : Off
 Max iterations          : 100
 KAIN solver             : 5
 Localization            : Off
 Diagonalization         : First two iterations
 Start precision         : 1.00000e-03
 Final precision         : 1.00000e-03
 Helmholtz precision     : Dynamic
 Energy threshold        : Off
 Orbital threshold       : 1.00000e-01
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


===========================================================================
 Iter           MO residual             Total energy                Update
---------------------------------------------------------------------------
    0          1.000000e+00          -7.222649094979         -7.222649e+00
    1          6.037085e-02          -7.320075664302         -9.742657e-02
---------------------------------------------------------------------------
                      SCF converged in 1 iterations!
===========================================================================


                                                                           
***************************************************************************
***                                                                     ***
***                    Printing Molecular Properties                    ***
***                                                                     ***
***************************************************************************
                                                                           
                                                                           
===========================================================================
                                 Molecule
---------------------------------------------------------------------------
 Charge                  :                                               1
 Multiplicity            :                                               1
---------------------------------------------------------------------------
    N    Atom            :               x               y               z
---------------------------------------------------------------------------
    0      Li            :        0.000000        0.000000        0.000000
---------------------------------------------------------------------------
 Center of mass          :        0.000000        0.000000        0.000000
===========================================================================


===========================================================================
                         Molecular Energy (final)
---------------------------------------------------------------------------
 Kinetic energy          :            (au)                  7.584212118792
 E-N energy              :            (au)                -16.491476051686
 Coulomb energy          :            (au)                  3.357678874268
 Exchange energy         :            (au)                 -0.419594145537
 X-C energy              :            (au)                 -1.286849061623
 Ext. field (el)         :            (au)                  0.000000000000
---------------------------------------------------------------------------
 N-N energy              :            (au)                  0.000000000000
 Ext. field (nuc)        :            (au)                  0.000000000000
 Tot. Reac. Energy       :            (au)                 -0.064047398516
 El. Reac. Energy        :            (au)                  0.128081335378
 Nuc. Reac. Energy       :            (au)                 -0.192128733894
---------------------------------------------------------------------------
 Electronic energy       :            (au)                 -7.127946930408
 Nuclear energy          :            (au)                 -0.192128733894
---------------------------------------------------------------------------
 Total energy            :            (au)             -7.320075664302e+00
                         :      (kcal/mol)             -4.593416793146e+03
                         :        (kJ/mol)             -1.921885587500e+04
                         :            (eV)             -1.991893974965e+02
===========================================================================


===========================================================================
                         Orbital Energies (final)
---------------------------------------------------------------------------
    n  Occ Spin          :                                         Epsilon
---------------------------------------------------------------------------
    0    2    p          :            (au)                 -2.210659505993
---------------------------------------------------------------------------
 Sum occupied            :            (au)                 -4.421319011986
===========================================================================


===========================================================================
                           Dipole Moment (dip-1)
---------------------------------------------------------------------------
 r_O                     :        0.000000        0.000000        0.000000
---------------------------------------------------------------------------
 Electronic vector       :       -0.000000        0.000000       -0.000000
 Magnitude               :            (au)                        0.000000
                         :         (Debye)                        0.000000
---------------------------------------------------------------------------
 Nuclear vector          :        0.000000        0.000000        0.000000
 Magnitude               :            (au)                        0.000000
                         :         (Debye)                        0.000000
---------------------------------------------------------------------------
 Total vector            :        0.000000        0.000000        0.000000
 Magnitude               :            (au)                        0.000000
                         :         (Debye)                        0.000000
===========================================================================


                                                                           
                                                                           
***************************************************************************
***                                                                     ***
***                            Exiting MRChem                           ***
***                                                                     ***
***                       Wall time :  0h  0m 48s                       ***
***                                                                     ***
***************************************************************************
                                                                           
                                                                           
